### DISCUSSION TITLE
[4.0] Frontend user is not logged out if browser window is closed

### DISCUSSION THREAD

**Discussion by Scrabble96** [joomla/joomla-cms#35999] [Q&A]
### Steps to reproduce the issue
Log in on the frontend e.g. on Firefox
Close the browser

### Expected result
User logged out

### Actual result
Open another browser window (same browser, e.g. Firefox, not Chrome)
User stays logged in.

### Additional comments
If this were a website holding financial/personal information, this could also be a security issue.

**Comment by simbus82** [comment]
Absolutely not. If you want the shortest login time just adjust the session time or browse in private/anonymous mode.
It is neither intended nor correct on the UX side to destroy a user's session if he closes the browser window. This is the normal behavior of a modern website.
If you want a security system such as when you access the customer area of a bank, you have to adapt everything there to your specific needs.

**Comment by laoneo** [comment]
Or write a plugin/template function which listens to the Javascript [onbeforeunload event ](https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload) and destroy the session cookie.

**Reply by Scrabble96** [reply]
Thanks, I'll have a look at that.

**Comment by chmst** [comment]
This is expected behaviour and normal in the web. I moved this to discussion - in case someone wants to follow the hint of @laoneo
